portfolio/src/main/java/com/google/sps/authentication/servlets/LoginServlet.java

```diff
@@ -10,19 +11,54 @@
 import java.io.IOException;
 
 @WebServlet("/login")
-public class LoginServlet extends HttpServlet {
+public final class LoginServlet extends HttpServlet {
 
     private UserService userService;
 
+    private class ResultType {
+        private boolean isLoggedIn;
+        private String htmlText;
+
+        private ResultType(boolean isLoggedIn, String htmlText) {
+            this.isLoggedIn = isLoggedIn;
+            this.htmlText = htmlText;
+        }
+    }
+
     @Override
     public void init() {
         this.userService = UserServiceFactory.getUserService();
     }
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        response.setContentType("text/html;");
+        response.setContentType("application/json;");
+
+        boolean isLoggedIn = this.userService.isUserLoggedIn();
+        String htmlText;
+
+        if (isLoggedIn) {
+            String userEmail = userService.getCurrentUser().getEmail();
+            String urlToRedirectToAfterUserLogsOut = "/index.html";
+            String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+            htmlText = String.format("<p>Hello %s!</p>\n"
+                    + "<p>Logout <a href=\"%s\">here</a>.</p>", userEmail, logoutUrl);
+        } else {
+            String urlToRedirectToAfterUserLogsIn = "/comments.html";
+            String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+            htmlText = String.format("<p>Hello stranger.</p>"
```

Comment:
Same as above, why are we dealing with html code here?

Reply:
Fixed